docs/Execution.md

```diff
@@ -0,0 +1,107 @@
+Execution
+======
+
+The MsQuic API uses a different execution model compared to BSD-style sockets and most other networking libraries built on them.
+The sections below detail the designs MsQuic uses and the reasons behind these choices.
+
+## Event Model
+
+In the MsQuic API, all state changes and notifications are indicated directly to the application via a callback.
+This includes connection state changes, new streams being created, stream data being received, and stream sends completing.
+
+```c
+typedef struct QUIC_LISTENER_EVENT {
+    QUIC_LISTENER_EVENT_TYPE Type;
+    union {
+        struct { ... } NEW_CONNECTION;
+        struct { ... } STOP_COMPLETE;
+    };
+} QUIC_LISTENER_EVENT;
+
+typedef
+_IRQL_requires_max_(PASSIVE_LEVEL)
+_Function_class_(QUIC_LISTENER_CALLBACK)
+QUIC_STATUS
+(QUIC_API QUIC_LISTENER_CALLBACK)(
+    _In_ HQUIC Listener,
+    _In_opt_ void* Context,
+    _Inout_ QUIC_LISTENER_EVENT* Event
+    );
+```
+
+Above is an example of a callback delivered to the listener interface.
+The application must register a callback handler to manage all events MsQuic may indicate, returning a status to show if it was successfully handled or not.
```

Comment:
nit: 'must register a callback handler for each type of event MsQuic may indicate" since this seems to suggest a single handler for all events.

Reply:
>must register a callback handler for each type of event MsQuic may indicate

☝️  this was my wording suggestion

Reply:
is this saying that there is a per-connection callback, and a per-stream callback, and a per-listener callback, etc., and then all events for that object type go through that callback? if so, i'd just write that.

Reply:
There is a per-object callback. Each callback handles all the types of events for that object. I will try to improve the text to make that clearer.